src/api_server/src/lib.rs

```diff
@@ -71,20 +79,73 @@ impl ApiServer {
     pub fn new(
         config: Config,
         sockets: Vec<SocketAddr>,
+        private_key: Option<PathBuf>,
+        certificate: Option<PathBuf>,
+        insecure: bool,
         attestation_service: Arc<AttestationService>,
         http_timeout: i64,
-    ) -> Self {
-        ApiServer {
+    ) -> Result<Self> {
+        if !insecure && (private_key.is_none() || certificate.is_none()) {
+            return Err(anyhow!("Missing HTTPS credentials"));
```

Comment:
NIT: we can use `anyhow::bail!("Missing HTTPS credentials")` to make shorter

Reply:
Thanks, fixed now.